packages/framework/agent-scheduler/src/scheduler.ts

```diff
@@ -267,10 +267,6 @@ export class AgentScheduler
 		// Probably okay for now to have every client try to do this.
 		// eslint-disable-next-line @typescript-eslint/no-misused-promises
 		quorum.on("removeMember", async (clientId: string) => {
-			assert(
```

Comment:
Should we be adding a test here just to verify what behavior we want to occur?

Reply:
I think the theoretical scenario would be:
1. Create a detached agent-scheduler on an attached container.
2. Locally sign up for a task
3. Get disconnected, causing your own client ID to be removed from the quorum
4. Something (currently undefined due to assert) happens to your local ownership of the task on the detached agent-scheduler.

Probably either want to explicitly blow up in this case ("don't register for tasks on a detached agent-scheduler") or else come up with a thought-through design for what should happen ("you can't lose a task on a detached agent-scheduler even if you disconnect" or "you lose the task on a detached agent-scheduler when you disconnect, just as if it were attached").  Probably don't want to implicitly support whatever happens by default from removing the assert, codifying the undefined behavior.

Since agent-scheduler is effectively on ice and not recommended to anyone, probably pick whichever is easiest/safest of the above options.

Reply:
Is there someone more familiar with agent scheduler that can take on that work? Since I'm OCE and have very little context on this, I would hesitate to make behavior changes.

Reply:
Considering agent scheduler is not recommended, maybe the safest thing to do is to simply return if data store is detached. That will keep the existing behavior without the assert. We can then create a bug to do the more involved fix.